Sources/SwiftParserDiagnostics/ParserDiagnosticMessages.swift

```diff
@@ -518,6 +518,23 @@ public struct UnknownDirectiveError: ParserError {
   }
 }
 
+public struct UnknownParameterError: ParserError {
+  public let parameter: TokenSyntax
+  public let validParameters: [TokenSyntax]
+
+  public var message: String {
+    var message = "unknown parameter '\(parameter.description)'"
+
+    if let parentTypeName = parameter.parent?.ancestorOrSelf(mapping: { $0.nodeTypeNameForDiagnostics(allowBlockNames: false) }) {
+      message += " in \(parentTypeName)"
+    }
+
+    message += "; valid parameters \(nodesDescription(validParameters, format: true))"
```

Comment:
Let’s make this a proper english sentence.

```suggestion
    message += "; valid parameters are \(nodesDescription(validParameters, format: true))"
```